jwst/scripts/world_coords.py

```diff
@@ -151,47 +160,51 @@ def build_output_files(filenames):
         outputs = []
 
     # Track if output file was a directory
-    if output is not None and os.path.isdir(output):
+    if output is not None and Path(output).is_dir():
         output_dir = output
     else:
         output_dir = None
 
     # Build an output filename for each input
     for filename in filenames:
         # Build output base and extensions
-        dirname, root = os.path.split(filename)
-        base, ext = os.path.splitext(root)
-        if ext != '.fits':
-            ext = '.fits'
-        if not base.endswith('world_coordinates'):
-            base = '_'.join([base, 'world_coordinates'])
+        path_object = Path(filename)
```

Comment:
I have similar potential concern about this refactor as the other major pathlib refactor.  How well-covered is this by tests?

Reply:
It is not covered by tests, but I feel more confident with this script because the code is easier to follow and the path changes are not as extensive.